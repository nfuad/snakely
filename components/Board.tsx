import * as React from 'react'

// custom components
import Snake from './Snake'
import Food from './Food'

export default (props) => (
  <div className="game-board">
    <Snake snakeDots={props.snakeDots} />
    <Food dot={props.food} />

    <style jsx global>{`
      .game-board {
        position: relative;
        margin: 50px auto;
        width: 30vw;
        height: 30vw;
        border: 2px solid #000;
      }
    `}</style>
  </div>
)
