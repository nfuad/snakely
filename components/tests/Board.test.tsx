import * as React from 'react'
import { shallow, mount } from 'enzyme'

import Board from '../Board'

const mockData = {
  food: [0, 1],
  snakeDots: [
    [0, 0],
    [2, 0]
  ]
}

describe('Board component', () => {
  it('renders', () => {
    shallow(
      <Board
        data={{
          food: mockData.food,
          snakeDots: mockData.snakeDots
        }}
      />
    )
  })

  it('matches the snapshot', () => {
    const component = mount(
      <Board
        data={{
          food: mockData.food,
          snakeDots: mockData.snakeDots
        }}
      />
    )
    expect(component).toMatchSnapshot()
  })

  it('updates via props', () => {
    const component = mount(
      <Board
        data={{
          food: mockData.food,
          snakeDots: mockData.snakeDots
        }}
      />
    )
    expect(component).toMatchSnapshot()

    component.setProps({
      food: [1, 0],
      snakeDots: [
        [0, 2],
        [2, 2]
      ]
    })

    expect(component).toMatchSnapshot()
  })
})
