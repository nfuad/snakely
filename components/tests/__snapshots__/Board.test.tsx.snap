// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Board component matches the snapshot 1`] = `
<_default
  data={
    Object {
      "food": Array [
        0,
        1,
      ],
      "snakeDots": Array [
        Array [
          0,
          0,
        ],
        Array [
          2,
          0,
        ],
      ],
    }
  }
>
  <div
    className="jsx-3514393862 paper game-board"
  >
    <_default
      snakeDots={
        Array [
          Array [
            0,
            0,
          ],
          Array [
            2,
            0,
          ],
        ]
      }
    >
      <div>
        <div
          className="jsx-1162698439 snake-dot"
          key="0"
          style={
            Object {
              "left": "0%",
              "top": "0%",
            }
          }
        >
          <JSXStyle
            id="1162698439"
          />
        </div>
        <div
          className="jsx-1162698439 snake-dot"
          key="1"
          style={
            Object {
              "left": "2%",
              "top": "0%",
            }
          }
        >
          <JSXStyle
            id="1162698439"
          />
        </div>
      </div>
    </_default>
    <_default
      dot={
        Array [
          0,
          1,
        ]
      }
    >
      <div
        className="jsx-2133318913 food"
      >
        <JSXStyle
          dynamic={
            Array [
              0,
              1,
            ]
          }
          id="1138619144"
        />
      </div>
    </_default>
    <JSXStyle
      id="3514393862"
    />
  </div>
</_default>
`;

exports[`Board component updates via props 1`] = `
<_default
  data={
    Object {
      "food": Array [
        0,
        1,
      ],
      "snakeDots": Array [
        Array [
          0,
          0,
        ],
        Array [
          2,
          0,
        ],
      ],
    }
  }
>
  <div
    className="jsx-3514393862 paper game-board"
  >
    <_default
      snakeDots={
        Array [
          Array [
            0,
            0,
          ],
          Array [
            2,
            0,
          ],
        ]
      }
    >
      <div>
        <div
          className="jsx-1162698439 snake-dot"
          key="0"
          style={
            Object {
              "left": "0%",
              "top": "0%",
            }
          }
        >
          <JSXStyle
            id="1162698439"
          />
        </div>
        <div
          className="jsx-1162698439 snake-dot"
          key="1"
          style={
            Object {
              "left": "2%",
              "top": "0%",
            }
          }
        >
          <JSXStyle
            id="1162698439"
          />
        </div>
      </div>
    </_default>
    <_default
      dot={
        Array [
          0,
          1,
        ]
      }
    >
      <div
        className="jsx-2133318913 food"
      >
        <JSXStyle
          dynamic={
            Array [
              0,
              1,
            ]
          }
          id="1138619144"
        />
      </div>
    </_default>
    <JSXStyle
      id="3514393862"
    />
  </div>
</_default>
`;

exports[`Board component updates via props 2`] = `
<_default
  data={
    Object {
      "food": Array [
        0,
        1,
      ],
      "snakeDots": Array [
        Array [
          0,
          0,
        ],
        Array [
          2,
          0,
        ],
      ],
    }
  }
  food={
    Array [
      1,
      0,
    ]
  }
  snakeDots={
    Array [
      Array [
        0,
        2,
      ],
      Array [
        2,
        2,
      ],
    ]
  }
>
  <div
    className="jsx-3514393862 paper game-board"
  >
    <_default
      snakeDots={
        Array [
          Array [
            0,
            0,
          ],
          Array [
            2,
            0,
          ],
        ]
      }
    >
      <div>
        <div
          className="jsx-1162698439 snake-dot"
          key="0"
          style={
            Object {
              "left": "0%",
              "top": "0%",
            }
          }
        >
          <JSXStyle
            id="1162698439"
          />
        </div>
        <div
          className="jsx-1162698439 snake-dot"
          key="1"
          style={
            Object {
              "left": "2%",
              "top": "0%",
            }
          }
        >
          <JSXStyle
            id="1162698439"
          />
        </div>
      </div>
    </_default>
    <_default
      dot={
        Array [
          0,
          1,
        ]
      }
    >
      <div
        className="jsx-2133318913 food"
      >
        <JSXStyle
          dynamic={
            Array [
              0,
              1,
            ]
          }
          id="1138619144"
        />
      </div>
    </_default>
    <JSXStyle
      id="3514393862"
    />
  </div>
</_default>
`;
